oauth2-server-core/src/main/java/nl/myndocs/oauth2/Oauth2TokenService.kt

```diff
@@ -293,7 +318,7 @@ class Oauth2TokenService(
     override fun userInfo(accessToken: String): UserInfo {
         val storedAccessToken = tokenStore.accessToken(accessToken) ?: throw InvalidGrantException()
         val client = clientService.clientOf(storedAccessToken.clientId) ?: throw InvalidClientException()
-        val identity = identityService.identityOf(client, storedAccessToken.username)
+        val identity = storedAccessToken.username?.let { identityService.identityOf(client, it) }
                 ?: throw InvalidIdentityException()
```

Comment:
Not sure about throwing an InvalidIdentityException here?  There is no identity, so it is invalid to try and get the identity, thoughts?

Reply:
The exception can be removed. It should not show the username key and value in that case

Reply:
Sorry I'm not completely following - as in the method should return `UserInfo?`?

Reply:
I forgot it was called `userInfo` I think it should be renamed to `tokenInfo` and make identity nullable on `TokenInfo` (what is now called `UserInfo`). 
In that case `tokenInfo` always provides all the information it haves. You can pick this up with this PR, otherwise I can pick this up outside of this PR.

Reply:
I think best to create a separate PR should be raised for this which I'm happy to do, I've created issue #32 to track work out the spec and track it